internal/bind/numeric_args.go

```diff
@@ -49,38 +52,23 @@ func (m NumericArgs) RewriteQuery(sql string, args ...interface{}) (
 			}
 			if int(p) > len(args) {
 				return "", nil, xerrors.WithStackTrace(
-					fmt.Errorf("%w: $p%d, len(args) = %d", ErrInconsistentArgs, p, len(args)),
+					fmt.Errorf("%w: $%d, len(args) = %d", ErrInconsistentArgs, p, len(args)),
 				)
 			}
-			paramName := "$p" + strconv.Itoa(int(p-1)) //nolint:goconst
-			if newArgs[p-1] == nil {
-				param, err = toYdbParam(paramName, args[p-1])
-				if err != nil {
-					return "", nil, xerrors.WithStackTrace(err)
-				}
-				newArgs[p-1] = param
-				buffer.WriteString(param.Name())
-			} else {
-				buffer.WriteString(newArgs[p-1].(table.ParameterOption).Name())
+			paramIndex := int(p - 1)
+			if newArgs[paramIndex] == nil {
```

Comment:
It will be never true in new code: newArgs fulled by `parsePositionalParameters` and if it has some error - the function will return earlier.